addon/PageWorker.js

```diff
@@ -0,0 +1,39 @@
+const {Page} = require("sdk/page-worker");
+const {data} = require("sdk/self");
+const {LOCAL_STORAGE_KEY} = require("common/constants");
+const {ADDON_TO_CONTENT} = require("common/event-constants");
+
+class PageWorker {
+  constructor({store}) {
+    this._page = null;
+    this._onDispatch = null;
+    this._unsubscribe = null;
+    this._store = store;
+  }
+  connect() {
+    this._page = Page({
+      contentURL: data.url("page-worker/page-worker.html"),
+      contentScriptFile: data.url("content-bridge.js"),
+      contentScriptWhen: "start"
+    });
+    this._onDispatch = () => {
+      this._page.port.emit(ADDON_TO_CONTENT, {
+        type: LOCAL_STORAGE_KEY,
+        data: this._store.getState()
+      });
+    };
+    this._unsubscribe = this._store.subscribe(this._onDispatch);
```

Comment:
#nit

```
To unsubscribe the change listener, invoke the function returned by subscribe.
```
That is __not obvious__ unless you go and read the docs. Can you please add a comment about this otherwise reading this line might seem very confusing.